frontend/src/components/common/Terminal.tsx

```diff
@@ -172,8 +172,11 @@ export default function Terminal(props: TerminalProps) {
       (async function () {
         send(4, `{"Width":${xterm.cols},"Height":${xterm.rows}}`);
       })();
-      xtermc.connected = true;
-      console.debug('Terminal is now connected');
+      // On server error, don't set it as connected
+      if (channel !== 3) {
```

Comment:
I don't understand how `channel !== 3` means not an error.

Reply:
There's a list of what channels mean, just above that block.

Reply:
Probably an enum would make this more readable.